tests/integration/test_cli.py

```diff
@@ -481,25 +488,27 @@ async def test_multipart_upload(
     # create big temp file
     monkeypatch.setattr(
         "ghga_connector.core.api_calls.well_knowns.WKVSCaller.get_ucs_api_url",
-        lambda x: api_url,
+        wkvs_method_mock(api_url),
     )
     with big_temp_file(file_size) as file:
         with patch(
             "ghga_connector.cli.CONFIG",
             get_test_config(),
         ):
             message_display = init_message_display(debug=True)
-            parameters = retrieve_upload_parameters()
-            await upload(
-                api_url=parameters.ucs_api_url,
-                file_id=file_id,
-                file_path=Path(file.name),
-                message_display=message_display,
-                server_public_key=parameters.server_pubkey,
-                my_public_key_path=Path(PUBLIC_KEY_FILE),
-                my_private_key_path=Path(PRIVATE_KEY_FILE),
-                part_size=DEFAULT_PART_SIZE,
-            )
+            async with async_client() as client:
+                parameters = await retrieve_upload_parameters(client=client)
+                await upload(
+                    api_url=parameters.ucs_api_url,
+                    client=client,
+                    file_id=file_id,
+                    file_path=Path(file.name),
+                    message_display=message_display,
+                    server_public_key=parameters.server_pubkey,
+                    my_public_key_path=Path(PUBLIC_KEY_FILE),
+                    my_private_key_path=Path(PRIVATE_KEY_FILE),
+                    part_size=DEFAULT_PART_SIZE,
+                )
```

Comment:
Could condense and dedent a smidge:
```suggestion
    with ( 
        big_temp_file(file_size) as file,
        patch("ghga_connector.cli.CONFIG", get_test_config()),
    ):
        message_display = init_message_display(debug=True)
        async with async_client() as client:
            parameters = await retrieve_upload_parameters(client=client)
            await upload(
                api_url=parameters.ucs_api_url,
                client=client,
                file_id=file_id,
                file_path=Path(file.name),
                message_display=message_display,
                server_public_key=parameters.server_pubkey,
                my_public_key_path=Path(PUBLIC_KEY_FILE),
                my_private_key_path=Path(PRIVATE_KEY_FILE),
                part_size=DEFAULT_PART_SIZE,
            )
```